rblib/format.rb

```diff
@@ -69,6 +69,14 @@ def self.make_clickable(text, params = {})
             end
             ret = ret.gsub(/(\s)([a-z0-9\-_.]+)@([^,< \n\r]*[^.,< \n\r])/i, "\\1<a href=\"mailto:\\2@\\3\">\\2@\\3</a>")
 
+            # If there are more closing brackets than opening brackets in the
+            # parsed link and the last char of the link is a closing bracket,
+            # remove the last closing bracket from the link
+            if ret =~ /\)<\/a>/ && ret =~ /href="([^"]+)"/ && ret.count(')') > ret.count('(')
+                ret = ret.gsub(/href="([^"]+)"/, "href=\"#{$1[0..-2]}\"")
+                ret = ret.gsub(")</a>", "</a>)")
+            end
```

Comment:
The following spec additions fail, so I'm not sure how robust a method this is:

```diff
diff --git a/rblib/tests/format_test.rb b/rblib/tests/format_test.rb
index 313cf81..605c08f 100644
--- a/rblib/tests/format_test.rb
+++ b/rblib/tests/format_test.rb
@@ -29,6 +29,16 @@ class TestFormat < Test::Unit::TestCase
     expected = 'The documentation (<a href="http://www.flourish.org/page(1)">' \
                'http://www.flourish.org/page(1)</a>) has the answer'
     expect_clickable(text, expected)
+
+    text = "The documentation (http://www.flourish.org/page(1))) has the answer"
+    expected = 'The documentation (<a href="http://www.flourish.org/page(1)">' \
+               'http://www.flourish.org/page(1)</a>)) has the answer'
+    expect_clickable(text, expected)
+
+    text = "The documentation ((http://www.flourish.org/page(1)) has the answer"
+    expected = 'The documentation ((<a href="http://www.flourish.org/page(1)">' \
+               'http://www.flourish.org/page(1)</a>) has the answer'
+    expect_clickable(text, expected)
   end

   def test_make_curly_brackets_clickable
```

I think the problem is more in the [initial parsing of URLs](http://rubular.com/r/LHrErwGYhb), since it picks up the closing brackets of the "text".

I wouldn't feel too bad about this – URI also gets it wrong:

```ruby
"The documentation (http://www.flourish.org/page(1)) has the answer".match(URI.regexp)[0]
# => "http://www.flourish.org/page(1))"
```

I think it would be worth having a go at it from this angle, but maybe limit it to half a day of exploration as the current commit does fix our immediate problem.